src/org/jetbrains/java/decompiler/modules/decompiler/SwitchPatternMatchProcessor.java

```diff
@@ -262,6 +266,41 @@ private static boolean processStatement(SwitchStatement stat, Statement root) {
       }
     }
 
+    Exprent oldSelector = realSelector;
+    // inline head
+    List<Exprent> basicHead = stat.getBasichead().getExprents();
+    if (realSelector instanceof VarExprent var && basicHead != null && basicHead.size() >= 1) {
+      if (basicHead.get(basicHead.size() - 1) instanceof AssignmentExprent assignment && assignment.getLeft() instanceof VarExprent assigned) {
+        if (var.equals(assigned) && !var.isVarReferenced(root, Stream.concat(Stream.of(assigned), stat.getCaseValues().stream().flatMap(List::stream).filter(exp -> exp instanceof FunctionExprent func && func.getFuncType() == FunctionType.INSTANCEOF && func.getLstOperands().get(0) instanceof VarExprent checked && checked.equals(var)).map(exp -> (VarExprent) ((FunctionExprent) exp).getLstOperands().get(0))).toArray(VarExprent[]::new))) {
+          realSelector = assignment.getRight();
+          basicHead.remove(basicHead.size() - 1);
+        }
+      }
+    }
+
+    // Check for non null
+    if (basicHead != null && basicHead.size() >= 1 && realSelector instanceof VarExprent var && !nullCase) {
+      Exprent last = basicHead.get(basicHead.size() - 1);
+      if (last instanceof InvocationExprent inv && inv.isStatic() && inv.getClassname().equals("java/util/Objects") && inv.getName().equals("requireNonNull") && inv.getStringDescriptor().equals("(Ljava/lang/Object;)Ljava/lang/Object;") && var.equals(inv.getLstParameters().get(0))) {
+        basicHead.remove(basicHead.size() - 1);
+        // Check for other assignment
+        if (basicHead.size() >= 1 && var.isStack() && !nullCase) {
+          last = basicHead.get(basicHead.size() - 1);
+          if (last instanceof AssignmentExprent assignment && assignment.getLeft() instanceof VarExprent assigned && var.equals(assigned)) {
+            if (!var.isVarReferenced(root, assigned)) {
+              realSelector = assignment.getRight();
+              basicHead.remove(basicHead.size() - 1);
+            }
+          }
+        }
+      }
+    }
+
+    if (oldSelector != realSelector) {
+      Exprent finalSelector = realSelector;
+      stat.getCaseValues().stream().flatMap(List::stream).filter(Objects::nonNull).filter(exp -> exp instanceof FunctionExprent func && func.getFuncType() == FunctionType.INSTANCEOF && func.getLstOperands().get(0).equals(oldSelector)).forEach(exp -> ((FunctionExprent) exp).getLstOperands().set(0, finalSelector));
```

Comment:
This line needs to be broken up into multiple lines, and it needs a comment explaining what it's trying to do.